DSCResources/MSFT_xVMHyperV/MSFT_xVMHyperV.psm1

```diff
@@ -200,14 +200,49 @@ function Set-TargetResource
             # Stop the VM, set the right properties, start the VM only if there are properties to change
             if ($changeProperty.Count -gt 0) {
                 Change-VMProperty -Name $Name -VMCommand "Set-VM" -ChangeProperty $changeProperty -WaitForIP $WaitForIP -RestartIfNeeded $RestartIfNeeded
+                Write-Verbose -Message "VM $Name updated"
             }
 
-            # If the VM does not have the right MACAddress, stop the VM, set the right MACAddress, start the VM
-            if($MACAddress -and ($vmObj.NetWorkAdapters.MacAddress -notcontains $MACAddress))
+            ## Set VM network switches. This can be done while the VM is running.
+            for ($i = 0; $i -lt $SwitchName.Count; $i++)
             {
-                Write-Verbose -Message "VM $Name does not have correct MACAddress. Expected $MACAddress, actual $($vmObj.NetWorkAdapters[0].MacAddress)"
-                Change-VMProperty -Name $Name -VMCommand "Set-VMNetworkAdapter" -ChangeProperty @{StaticMacAddress=$MACAddress} -WaitForIP $WaitForIP -RestartIfNeeded $RestartIfNeeded
-                Write-Verbose -Message "VM $Name now has correct MACAddress."
+                $switch = $SwitchName[$i]
+                $nic = $vmObj.NetworkAdapters[$i]
+                if ($nic) {
+                    ## We cannot change the MAC address whilst the VM is running..
```

Comment:
@iainbrighton Never mind I missed some of the code in the first review.  Suggestion, I think it would make it more clear if you move the comment on line 212 to around line 243.